src/preact.d.ts

```diff
@@ -51,19 +66,24 @@ declare namespace preact {
 		static displayName?:string;
 		static defaultProps?:any;
 
-		state:StateType;
-		props:PropsType & ComponentProps<this>;
+		state:Readonly<StateType>;
+		props:Readonly<PropsType> & Readonly<{children?: JSX.Element[]}>;
 		context:any;
 		base:HTMLElement;
 
+		/**
+		 * @deprecated
+		 *
+		 * Not part of the core preact anymore
+		 */
 		linkState:(name:string) => (event: Event) => void;
```

Comment:
Considering the typings are shipped with Preact itself, this can just be removed